clients/src/main/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinator.java

```diff
@@ -689,7 +690,6 @@ protected void onJoinPrepare(int generation, String memberId) {
         }
 
         isLeader = false;
-        subscriptions.resetGroupSubscription();
```

Comment:
I guess we could restore this location with the change to the group subscription maintenance? I don't feel too strongly since the new location works also.

Reply:
@hachikuji Thanks for the review. Restored this so that the only change from this PR is the clean separation of group and local subscriptions